SCHEMAS.sql

```diff
@@ -0,0 +1,304 @@
+    NAME VARCHAR(50),
+    EMAIL VARCHAR(50) UNIQUE,
+    PHONENUMBER VARCHAR(13),
+    PASSWORD VARCHAR(12),
+    REGISTRATIONDATE DATETIME DEFAULT CURRENT_TIMESTAMP
+  );
+
+CREATE TABLE MOVIES(
+    MOVIEID INT IDENTITY(1,1) PRIMARY KEY,
+    MOVIENAME VARCHAR(200),
+    MOVIEGENRE VARCHAR(100),
+    DURATION TIME,
+    LANGUAGE VARCHAR(50),
+    RELEASEDATE DATE,
+    RATING DECIMAL(3,1),
+    DESCRIPTION TEXT
+);
+
+CREATE TABLE THEATRES(
+    THEATREID INT IDENTITY(1,1) PRIMARY KEY,
+    NAME VARCHAR(200),
+    CITY VARCHAR(100),
+    STATE VARCHAR(100),
+    ZIPCODE VARCHAR(30),
+    TOTALSCREENS INT,
+    PHONENUMBER VARCHAR(13)
+);
+
+CREATE TABLE SCREENS (
+    SCREENID INT IDENTITY(1,1) PRIMARY KEY ,
+    THEATREID INT,
+    SCREENNUMBER INT,
+    SEATINGCAPACITY INT,
+    FOREIGN KEY (THEATREID) REFERENCES THEATRES(THEATREID)
+);
+
+CREATE TABLE SHOW (
+    SHOWID INT IDENTITY(1,1) PRIMARY KEY ,
+    MOVIEID INT,
+    SCREENID INT,
+    SHOWDATETIME DATETIME,
+    AVAILABLESEATS INT,
+    FOREIGN KEY (MOVIEID) REFERENCES MOVIES(MOVIEID),
+    FOREIGN KEY (SCREENID) REFERENCES SCREENS(SCREENID)
+);
+
+CREATE TABLE BOOKINGS (
+    BOOKINGID INT IDENTITY(1,1) PRIMARY KEY ,
+    USERID INT,
+    SHOWTIMEID INT,
+    BOOKINGDATE DATETIME DEFAULT CURRENT_TIMESTAMP,
+    TOTALAMOUNT DECIMAL(10, 2),
+    FOREIGN KEY (USERID) REFERENCES USERS(USERID),
+    FOREIGN KEY (SHOWTIMEID) REFERENCES SHOW(SHOWID)
+);
+
+CREATE TABLE TICKETS (
+    TICKETID INT IDENTITY(1,1) PRIMARY KEY,
+    BOOKINGID INT,
+    SEATNUMBER VARCHAR(10),
+    PRICE DECIMAL(10, 2),
+    FOREIGN KEY (BOOKINGID) REFERENCES BOOKINGS(BOOKINGID)
+);
+
+CREATE TABLE SEAT (
+    SEATID INT IDENTITY(1,1) PRIMARY KEY,
+    SHOWID INT,
+    SEATNUMBER VARCHAR(10),
+    SEATTYPE VARCHAR(20),
+    PRICE DECIMAL(10, 2),
+    FOREIGN KEY (SHOWID) REFERENCES SHOW(SHOWID)
+);
+
+CREATE TABLE BOOKEDSEAT (
+    BOOKINGID INT,
+    SEATID INT,
+    PRIMARY KEY (BOOKINGID, SEATID),
+    FOREIGN KEY (BOOKINGID) REFERENCES BOOKINGS(BOOKINGID),
+    FOREIGN KEY (SEATID) REFERENCES SEAT(SEATID)
+);
+
+------------------------------------------------------------------------------------------------------------
+
+"SCHEMA FOR AMAZON"
+
+  CREATE TABLE Users (
+    UserID INT PRIMARY KEY IDENTITY(1,1),
+    Name VARCHAR(100),
+    Email VARCHAR(100) UNIQUE,
+    PhoneNumber VARCHAR(15),
+    Password VARCHAR(100),
+    RegistrationDate DATETIME DEFAULT CURRENT_TIMESTAMP
+);
+
+CREATE TABLE Addresses (
+    AddressID INT PRIMARY KEY IDENTITY(1,1),
+    UserID INT,
+    Street VARCHAR(100),
+    City VARCHAR(100),
+    State VARCHAR(100),
+    ZipCode VARCHAR(20),
+    Country VARCHAR(100),
+    FOREIGN KEY (UserID) REFERENCES Users(UserID)
+);
+
+CREATE TABLE Categories (
+    CategoryID INT PRIMARY KEY identity(1,1),
+    CategoryName VARCHAR(100)
+);
+
+CREATE TABLE Products (
+    ProductID INT PRIMARY KEY IDENTITY(1,1),
+    Name VARCHAR(100),
+    Description TEXT,
+    Price DECIMAL(10, 2),
```

Comment:
Evaluate if DECIMAL(10,2) is sufficient for all price scenarios

Reply:
DECIMAL(10,2) is valid for typical price scenarios.
MONEY can also be used.